File: modules/expo-storage/ios/StorageCore.m
#import "StorageCore.h"

@interface StorageCore()
@property (nonatomic, strong) NSMutableDictionary<NSString *, NSString *> *storage;
@end

@implementation StorageCore

+ (instancetype)shared {
    static StorageCore *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (instancetype)init {
    if (self = [super init]) {
        _storage = [NSMutableDictionary dictionary];
        NSLog(@"🚀 [StorageCore] Initialized (Pure native, no RN dependencies)");
    }
    return self;
}

// ========================================
// BUSINESS LOGIC
// ========================================

- (void)setItem:(NSString *)key value:(NSString *)value {
    NSLog(@"📦 [StorageCore] setItem: '%@' = '%@'", key, value);
    self.storage[key] = value;
}

- (NSString *)getItem:(NSString *)key {
    return self.storage[key];
}

- (void)removeItem:(NSString *)key {
    [self.storage removeObjectForKey:key];
}

- (NSArray<NSString *> *)getAllKeys {
    return [self.storage allKeys];
}

- (void)clear {
    [self.storage removeAllObjects];
    NSLog(@"🔄 [StorageCore] Storage cleared");
}

@end

File: modules/expo-storage/ios/StorageCore.swift
import Foundation

/**
 * StorageCore - Pure native business logic (NO React Native dependencies)
 * 
 * This class owns the storage state and contains all storage logic.
 * It can be called from:
 * - ExpoStorageModule (Expo Module for JavaScript)
 * - ModuleInterop (for other native modules if needed)
 * - Unit tests (no RN runtime needed)
 */
public class StorageCore {
    public static let shared = StorageCore()
    
    private var storage: [String: String] = [:]
    
    private init() {
        // Initialize with empty storage
    }
    
    // ========================================
    // BUSINESS LOGIC
    // ========================================
    
    public func setItem(_ key: String, value: String) {
        storage[key] = value
    }
    
    public func getItem(_ key: String) -> String? {
        return storage[key]
    }
    
    public func removeItem(_ key: String) {
        storage.removeValue(forKey: key)
    }
    
    public func getAllKeys() -> [String] {
        return Array(storage.keys)
    }
    
    public func clear() {
        storage.removeAll()
    }
}

